nixos/modules/services/web-apps/weblate.nix

```diff
@@ -0,0 +1,422 @@
+      DEBUG = False
+
+      DATABASES = {
+        "default": {
+          "ENGINE": "django.db.backends.postgresql",
+          "HOST": "/run/postgresql",
+          "NAME": "weblate",
+          "USER": "weblate",
+        }
+      }
+
+      with open("${cfg.djangoSecretKeyFile}") as f:
+        SECRET_KEY = f.read().rstrip("\n")
+
+      CACHES = {
+        "default": {
+          "BACKEND": "django_redis.cache.RedisCache",
+          "LOCATION": "unix://${config.services.redis.servers.weblate.unixSocket}",
+          "OPTIONS": {
+              "CLIENT_CLASS": "django_redis.client.DefaultClient",
+              "PASSWORD": None,
+              "CONNECTION_POOL_KWARGS": {},
+          },
+          "KEY_PREFIX": "weblate",
+          "TIMEOUT": 3600,
+        },
+        "avatar": {
+          "BACKEND": "django.core.cache.backends.filebased.FileBasedCache",
+          "LOCATION": "/var/lib/weblate/avatar-cache",
+          "TIMEOUT": 86400,
+          "OPTIONS": {"MAX_ENTRIES": 1000},
+        }
+      }
+
+
+      CELERY_TASK_ALWAYS_EAGER = False
+      CELERY_BROKER_URL = "redis+socket://${config.services.redis.servers.weblate.unixSocket}"
+      CELERY_RESULT_BACKEND = CELERY_BROKER_URL
+
+      VCS_BACKENDS = ("weblate.vcs.git.GitRepository",)
+
+    ''
+    + lib.optionalString cfg.smtp.enable ''
+      ADMINS = (("Weblate Admin", "${cfg.smtp.user}"),)
+
+      EMAIL_HOST = "${cfg.smtp.host}"
+      EMAIL_USE_TLS = True
+      EMAIL_HOST_USER = "${cfg.smtp.user}"
+      SERVER_EMAIL = "${cfg.smtp.user}"
+      DEFAULT_FROM_EMAIL = "${cfg.smtp.user}"
+      EMAIL_PORT = 587
+      with open("${cfg.smtp.passwordFile}") as f:
+        EMAIL_HOST_PASSWORD = f.read().rstrip("\n")
+
+    ''
+    + cfg.extraConfig;
+  settings_py =
+    pkgs.runCommand "weblate_settings.py"
+      {
+        inherit weblateConfig;
+        passAsFile = [ "weblateConfig" ];
+      }
+      ''
+        mkdir -p $out
+        cat \
+          ${finalPackage}/${python.sitePackages}/weblate/settings_example.py \
+          $weblateConfigPath \
+          > $out/settings.py
+      '';
+  uwsgiConfig.uwsgi = {
+    type = "normal";
+    plugins = [ "python3" ];
+    master = true;
+    socket = "/run/weblate.socket";
+    die-on-idle = true;
+    die-on-term = true;
+    idle = 600;
+    manage-script-name = true;
+    cheap = true;
+    chmod-socket = "770";
+    chown-socket = "weblate:weblate";
+    uid = "weblate";
+    gid = "weblate";
+    wsgi-file = "${finalPackage}/${python.sitePackages}/weblate/wsgi.py";
+
+    # Some more recommendations by upstream:
+    # https://docs.weblate.org/en/latest/admin/install.html#sample-configuration-for-nginx-and-uwsgi
+    buffer-size = 8192;
+    reload-on-rss = 250;
+    workers = 8;
+    enable-threads = true;
+    close-on-exec = true;
+    umask = "0022";
+    ignore-sigpipe = true;
+    ignore-write-errors = true;
+    disable-write-exception = true;
+  };
```

Comment:
Is this basically **the** configuration to be used with weblate, or would people want to change this when they configure the module? Right now it's quite a pain to override.

Reply:
Actually only the lower part after the comment are actual recommendations by upstream, the upper part is mostly stuff that I set and then forgot about why I did it :see_no_evil: 

I agree with your concerns about overridability. I also looked a bit into uwsgi and realised the project is in maintenance mode for some time now. Maybe it just makes sense to switch to gunicorn, as you proposed [here](https://github.com/NixOS/nixpkgs/pull/325541#discussion_r1695185961).

Reply:
Using gunicorn instead worked great!